### DISCUSSION TITLE
Unable to install package PlatformIO

### DISCUSSION THREAD

**Discussion by LocoMatt** [SpenceKonde/megaTinyCore#657] [Library discussion]
Not much to say, I tried installing megaTinyCore as described [here](https://registry.platformio.org/tools/platformio/framework-arduino-megaavr-megatinycore/installation).
When I run 'build', PlatformIO returns:
```
Platform Manager: Installing platformio/framework-arduino-megaavr-megatinycore @ ^2.5.10
Error: Could not find the package with 'platformio/framework-arduino-megaavr-megatinycore @ ^2.5.10' requirements for your system 'windows_amd64'
```
I tried also with older versions without succeed. 

```
[env:test]
platform = platformio/framework-arduino-megaavr-megatinycore@^2.5.10
board = ATtiny827
framework = arduino
```

Looking forward any suggestions how to solve this.

**Comment by SpenceKonde** [comment]
I don't know enough about PIO to answer this. I don't know what it's missing or whether it's even a problem that's in my domain to fix or must be fixed by some platformIO person.

**Comment by LocoMatt** [comment]
Thanks for response, will try to ask about it on PIO forums. That's bit odd to me. If I'll get any useful information, I'll leave it here.

**Comment by br74649** [comment]
hi,

Hi @LocoMatt  

I don't know how far you got with this, but i had the same problems with the attiny1624 chips, they are not quite there with the integration of avr2's

recently @MCUdude did some work and added support to the new boards in the #develop branch and its working 

to do it for me, and get udpi working on serial i did the following

in platformio.ini set this

[env:ATtiny1624]
platform = https://github.com/platformio/platform-atmelmegaavr#develop
board = ATtiny1624
framework = arduino
upload_protocol = custom 
upload_command = C:\Users\tony_\AppData\Local\Arduino15\packages\megaTinyCore\tools\python3\3.7.2-post1/python3 -u C:\Users\tony_\AppData\Local\Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.10/tools/prog.py -t uart -u COM18 -b 230400 -d attiny1624 --fuses 2:0x01 6:0x04 8:0x00 -f$SOURCE -a write -v 
build_flags = 
  -DCORE_ATTACH_OLD

after saving that i had to go into the .platformio directories and delete some packages 

for me it was here

C:\Users\tony_\.platformio\packages

i then deleted the following directories

toolchain-atmel-avr
framework-arduino-megaavr-megatinycore
tool-avrdude-megaavr

and under C:\Users\tony_\.platformio\platforms 

i deleted 

atmelmegaavr

once thats done do a clean

and when you build it redownloads it all installs the 2.5.10 core

its working for me

in platformio.ini  

build_flags = 
  -DCORE_ATTACH_OLD

was suggested by @SpenceKonde  as i was getting strange issues with the timers in attachinterrupt and that seems to fix it

if you are not using an ftdi for udpi programming then the upload_command  and upload_protocol can be omitted but if you are, change the chip to whatever one you are using and set the serial com port to the correct one

hopefully this works for you too and saves you a weekend

Regards

Tony

**Comment by SpenceKonde** [comment]
Speaking of CORE_ATTACH_OLD I would *love* to see someone with a sketch that was fixed by that, and retest using the current github version of WInterrupts.c without that define. It should work finally, and use less memory and execute faster than the old one As soon as this i get omeone reporting that it didn't work w/out CORE_ATTACH_OLD with released WInterrupts.c and does work with github version, I'll feel a lot better about releasing the fix. These problems have just gone on so long, and I have no faith in my ability to determine if the fixes work anymrore, after being wrong 10 or or so times.

**Reply by br74649** [reply]
very happy to oblige, i can try it out and report back 

ill come back here if i cannot figure out how, but i got the sketch/hardware and scenario so its a start!

Thanks 

Tony
